From: Jo Zzsi <jozzsicsataban@gmail.com>
Date: Sat, 30 Aug 2025 21:45:52 -0400
Subject: fix(systemd-sysusers): increase ordering from 68 to 78

This module should be orders afer all modules that depends on it

This is to make sure that all inst_sysusers calls are in place
before systemd-sysusers is called.

This rule is already documented in the module-setup.sh file
but was not followed for all modules.

Fixes: https://github.com/dracut-ng/dracut-ng/issues/1610
Origin: upstream, https://github.com/dracut-ng/dracut-ng/pull/1613
---
 modules.d/68systemd-sysusers/module-setup.sh | 31 ----------------------------
 modules.d/78systemd-sysusers/module-setup.sh | 31 ++++++++++++++++++++++++++++
 2 files changed, 31 insertions(+), 31 deletions(-)
 delete mode 100755 modules.d/68systemd-sysusers/module-setup.sh
 create mode 100755 modules.d/78systemd-sysusers/module-setup.sh

diff --git a/modules.d/68systemd-sysusers/module-setup.sh b/modules.d/68systemd-sysusers/module-setup.sh
deleted file mode 100755
index 477b8e7..0000000
--- a/modules.d/68systemd-sysusers/module-setup.sh
+++ /dev/null
@@ -1,31 +0,0 @@
-#!/bin/bash
-# This file is part of dracut.
-# SPDX-License-Identifier: GPL-2.0-or-later
-
-# This module should be orders afer all modules that depends on it
-# This is to make sure that all inst_sysusers calls are in place before systemd-sysusers is called.
-
-# Prerequisite check(s) for module.
-check() {
-    # If the binary(s) requirements are not fulfilled the module can't be installed.
-    require_binaries systemd-sysusers || return 1
-
-    # Return 255 to only include the module, if another module requires it.
-    return 255
-}
-
-# Install the required file(s) and directories for the module in the initramfs.
-install() {
-    inst_sysusers basic.conf
-
-    # redirect stdout temporarily to FD 3 to use filter stderr
-    {
-        set -o pipefail
-        systemd-sysusers --root="$initdir" 2>&1 >&3 | grep -v "^Creating " >&2
-    } 3>&1
-
-    # systemd-sysusers does not set any permission
-    # set read and write permission for the current user
-    [[ -f "$initdir/etc/gshadow" ]] && chmod u+rw "$initdir/etc/gshadow"
-    [[ -f "$initdir/etc/shadow" ]] && chmod u+rw "$initdir/etc/shadow"
-}
diff --git a/modules.d/78systemd-sysusers/module-setup.sh b/modules.d/78systemd-sysusers/module-setup.sh
new file mode 100755
index 0000000..477b8e7
--- /dev/null
+++ b/modules.d/78systemd-sysusers/module-setup.sh
@@ -0,0 +1,31 @@
+#!/bin/bash
+# This file is part of dracut.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# This module should be orders afer all modules that depends on it
+# This is to make sure that all inst_sysusers calls are in place before systemd-sysusers is called.
+
+# Prerequisite check(s) for module.
+check() {
+    # If the binary(s) requirements are not fulfilled the module can't be installed.
+    require_binaries systemd-sysusers || return 1
+
+    # Return 255 to only include the module, if another module requires it.
+    return 255
+}
+
+# Install the required file(s) and directories for the module in the initramfs.
+install() {
+    inst_sysusers basic.conf
+
+    # redirect stdout temporarily to FD 3 to use filter stderr
+    {
+        set -o pipefail
+        systemd-sysusers --root="$initdir" 2>&1 >&3 | grep -v "^Creating " >&2
+    } 3>&1
+
+    # systemd-sysusers does not set any permission
+    # set read and write permission for the current user
+    [[ -f "$initdir/etc/gshadow" ]] && chmod u+rw "$initdir/etc/gshadow"
+    [[ -f "$initdir/etc/shadow" ]] && chmod u+rw "$initdir/etc/shadow"
+}
